src/mdoc/index.ts

```diff
@@ -1,28 +1,36 @@
-import { CBOR } from "@pagopa/io-react-native-cbor";
+import { CBOR, COSE } from "@pagopa/io-react-native-cbor";
 import type { JWK } from "../utils/jwk";
+import type { PublicKey } from "@pagopa/io-react-native-crypto";
+import { b64utob64 } from "jsrsasign";
+import { derToPem, getSigningJwk, parsePublicKey } from "../utils/crypto";
 
 export const verify = async (
   token: string,
-  publicKey: JWK | JWK[]
-): Promise<{ mDoc: CBOR.MDOC }> => {
+  _: JWK | JWK[]
+): Promise<{ issuerSigned: CBOR.IssuerSigned }> => {
   // get decoded data
-  const documents = await CBOR.decodeDocuments(token);
-  if (!documents || documents.documents.length === 0) {
-    throw new Error("Invalid mDoc");
-  }
-  const mDoc = documents.documents[0];
-  if (!mDoc) {
+  const issuerSigned = await CBOR.decodeIssuerSigned(token);
+  if (!issuerSigned) {
     throw new Error("Invalid mDoc");
   }
 
-  const sigKey = Array.isArray(publicKey)
-    ? publicKey.find((k) => k.use === "sig")
-    : publicKey;
-  sigKey;
+  const cert = issuerSigned.issuerAuth.unprotectedHeader[0]?.keyId;
+  if (!cert) throw new Error("Certificate not present in credential");
+
+  const pemcert = derToPem(b64utob64(cert));
+  const publickey = parsePublicKey(pemcert);
+  if (!publickey) throw new Error("Certificate not present in credential");
```

Comment:
ditto. Maybe I don't have the right context but looking at the code it seems that it is only converting the key id. Are you sure it has to work this way? I suggest adding more unit-tests on this